ToDoApp/ToDoAppTests/ToDoServiceTests.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ToDoServiceTests.swift
+//  ToDoAppTests
+//
+//  Created by Trong Hieu Vuong on 2022-12-30.
+//
+import XCTest
+@testable import ToDoApp
+
+final class RepositoryMock<Value>: Repository where Value: Identifiable, Value: Encodable, Value.ID == UUID {
+    private let _createOrUpdate: (Value) -> RepositoryError?
+    
+    init(createOrUpdate: @escaping (Value) -> RepositoryError?) {
+        self._createOrUpdate = createOrUpdate
+    }
+    
+    func createOrUpdate<T>(_ value: T) async -> RepositoryError? where T: Identifiable, T: Encodable, T.ID == UUID {
+        _createOrUpdate(value as! Value)
+    }
+    func read<T>() async ->Result<[T], RepositoryError> where T: Decodable {
+        fatalError()
+    }
+}
+final class ToDoServiceTests: XCTestCase {
+
+    func testCreateTask_Success() async {
```

Comment:
yeah that's the example test case. can you now implement testing create task failure scenario? :)

Reply:
Yes, I'll try